fix: ⚡️ Fix dependency vulnerabilities

melMass opened the pull request:
## Disclaimer

This is updating a lot of dependencies. To properly test it locally if you have already pulled the repo before, you must first delete `node_modules` and re-run `npm i`

---
Current `main`:
```ruby
50 vulnerabilities (27 moderate, 20 high, 3 critical)
```

This PR:
```ruby
10 vulnerabilities (7 moderate, 3 high)
```

---
I'm following dependabot alerts for now.
Most of them actually came from the templates.

Fixes:
[dependabot/2](https://github.com/teia-community/teia-ui/security/dependabot/2)
[dependabot/9](https://github.com/teia-community/teia-ui/security/dependabot/9)
[dependabot/50](https://github.com/teia-community/teia-ui/security/dependabot/50)
[dependabot/26](https://github.com/teia-community/teia-ui/security/dependabot/26)

melMass committed:
fix: ⚡️ Fix dependency vulnerabilities

I'm following dependabot alerts for now.
Most of them actually came from the templates.

melMass committed:
fix: ⚡️ use a unique id if ophash is undefined

melMass committed:
fix: ⚡️ use an <li> to fix the key error

melMass committed:
fix: ⚡️ Updating deps, trying to replace eztz-lib

Eztz-lib is used to verify the sign status of collab contracts I think.
It comes with a lot of critical security holes in their dependencies.

melMass committed:
feat: ⬆️ updating to react-scripts 5+

react-scripts 5 is now using WebPack5 in which there is no polyfills
by default anymore.

There is a PR on their repo about adding them back.
For now I propose to use craco, which is a way to extend react-scripts
without ejecting it. This allows us to extend the webpack config
and add polyfills.

melMass commented on .env:
```diff
@@ -1,3 +1,5 @@
+GENERATE_SOURCEMAP=false
```
This is due to that: 
https://github.com/facebook/create-react-app/discussions/11767

melMass commented on src/utils/verify.js:
```diff
@@ -0,0 +1,27 @@
+var sodium = require('sodium-universal')
```
I extracted the functions we make us of only from the ez lib thing which had many security issues associated. 

This seems to still error out. I need to investigate further

melMass replied:
Solved in 2ec678975c6f9ca07d77dc4ef4e72db6b8725462

melMass committed:
fix: ⚡️ use id instead of ophash

The indexer returns duplicates, but id's are unique.

melMass committed:
fix: ⚡️ key values for tags

melMass committed:
fix: 🔥 Using aliases

- Adds aliases.
- Also moves some of the common methods in search, index and galleries

melMass committed:
fix: ⚡️ get random objkts

This fixes #26

melMass committed:
fix: 🚑️ use libsodium-wrappers

melMass commented:
Ready to review.
IMO we should merge it and not Squash it because there is a lot going on, I tried to separate the commits properly

melMass commented:
@xat Do you know why there is no action running anymore?

xat commented:
hmm, the action did work in the pr that i just opened. maybe a temp problem?

xat commented:
@melMass i would suggest that we wait a week or so, before merging this one after we ironed out the more pressing bugs that got revealed due to the bigger release. because updating dependencies can bring in further instability, which i would want to avoid until things are bit more settled.

would it make sense to move the "get random objkts" commit into a separate pr, then we could merge that one upfront?

what do you think?

melMass commented:
> @melMass i would suggest that we wait a week or so, before merging this one after we ironed out the more pressing bugs that got revealed due to the bigger release.


Yes it makes sense, I spent hours on it so I won't have the same time to dedicate fixing it in a week, but we could use it as a reference or something like that 👌

xat commented:
@melMass ah ok, so delaying the merge would create extra effort.

hmm, do you think it is safe enough to merge it sooner? my feeling is that it does require some extended testing.

melMass commented:
> @melMass ah ok, so delaying the merge would create extra effort.
> 
> 
> 
> hmm, do you think it is safe enough to merge it sooner? my feeling is that it does require some extended testing.

Yes it definitely need more extensive testing, but I meant more if we fix other stuffs under it it might be better to start this one over, following it as a ref. I spent most of my time checking breaking changes in libraries, I know them now

melMass commented:
Rebased 😏  🤣

I'm going to try to maintain it as a virtual `develop` branch and rebase regularly until someone can look into it/ we can test it further

xat commented on craco.config.js:
```diff
@@ -0,0 +1,33 @@
+const path = require("path");
+module.exports = {
+  webpack: {
+    alias: {
```
nice 👍

xat commented on package.json:
```diff
@@ -3,49 +3,48 @@
+    "@airgap/beacon-sdk": "^2.3.10",
+    "@taquito/beacon-wallet": "^12.0.0",
+    "@taquito/michelson-encoder": "^12.0.0",
+    "@taquito/taquito": "^12.0.0",
```
did you check if there were any breaking changes?

melMass replied:
Yep that's the whole idea of this PR

xat commented on package.json:
```diff
@@ -3,49 +3,48 @@
-    "node-sass": "^6.0.1",
+    "markdown-to-jsx": "^7.1.7",
+    "mime-types": "^2.1.35",
+    "nanoassert": "^2.0.0",
```
is this used?

melMass replied:
I'm not sure like that I'm going to check in context!

xat commented on package.json:
```diff
@@ -66,8 +65,17 @@
-    "prettier": "^2.3.0",
-    "husky": "^7.0.0"
+    "@craco/craco": "^7.0.0-alpha.3",
+    "assert": "^2.0.0",
```
why were these libraries not needed before?

melMass replied:
[Craco](https://github.com/gsoft-inc/craco) is what allows all I'm doing here: extend CRA without ejecting

melMass replied:
Assert for the polyfills, I'm not 100% sure it's needed but if not used I don't think it adds anything to the bundle

xat commented on src/components/collab/create/CollaboratorRow.js:
```diff
@@ -1,11 +1,11 @@
-import { Button, Secondary } from '../../../components/button'
-import {} from '../../media-types/'
+import { Button, Secondary } from '@components/button'
+import {} from '@components/media-types/'
```
import can be removed

xat commented on src/data/hicdex.js:
```diff
@@ -207,3 +209,62 @@ export async function fetchGraphQL(operationsDoc, operationName, variables) {
+export async function fetchRandomObjkts(count) {
+  const firstId = 196
+  const lastId = await getLastObjktId()
+  console.log(`Last id is : ${lastId}`)
```
console.logs in the file probably can be removed

melMass replied:
I'm going to do a round of console.log indeed! I've seen others while checking the console.

melMass replied:
Mmm I'll just remove those you're right, it will complicate rebases on top. But I think we can remove 90% of them 😅

xat commented on src/data/hicdex.js:
```diff
@@ -207,3 +209,62 @@ export async function fetchGraphQL(operationsDoc, operationName, variables) {
+}
+
+export async function fetchRandomObjkts(count) {
+  const firstId = 196
```
unrelated to this pr, but afaik 152 is the first objkt id

melMass replied:
As far as I can tell this value was always set to that. I wonder what is between 152 and 196 maybe tests?

xat commented on src/data/hicdex.js:
```diff
@@ -207,3 +209,62 @@ export async function fetchGraphQL(operationsDoc, operationName, variables) {
+    let objkts = await fetchObjkts(Array.from(uniqueIds))
+    return objkts
+  } catch (e) {
+    console.error(e)
```
would it make sense to re-throw the error here?

melMass replied:
I think one of the refactor aspects will be to standardize that. Some return undefined, some log errors, some throw!
I agree they should all throw and excpetions should be handled where it's used

xat committed:
use opid as key of swaps in history tab

xat committed:
fix swap link in history tab

xat commented:
looks very nice!

melMass committed:
fix: ⚡️ edit dependencies

* Moved polyfills in devDependencies.
* Matching beacon-sdk's version of the libsodium-wrappers
* Update the lockfile

melMass committed:
fix: 🔥 the polyfills

* 90% of the issue was because of ES6 "import" instead of "require"
  for the verify function.
* 10% remains because Webpack 5 does not polyfills node built-ins
  (e.g. Buffer) anymore.

melMass commented:
Ok now it's ready to be reviewed for real @xat 😇

xat commented:
from the first review, there are few (small) open questions, but otherwise it looks good!

maybe some non-devs can do a bit of manual testing before we merge it.

melMass committed:
fix: 🎨 infinite scroll on main feed

melMass commented:
@xat Mmhh so while debugging the feed thing (that's actually search), it seems src/pages/feed isn't used anymore?

xat commented:
> @xat Mmhh so while debugging the feed thing (that's actually search), it seems src/pages/feed isn't used anymore?

yeah, looks like `/random`, `/latest` and `/hdao` are dead routes

melMass committed:
fix: ⚡️ the initial value of beneficiaries

The first parameter is the react comp itself when used as a callback.

We decided to not fix more in this PR as it's not related to the update.

xat reviewed (approved):
can be merged imo, if enough testing was done

xat commented:
nice work :>

melMass committed:
fix: 🎨 cleanup

* Commented out unused polyfills (they might be used in the future)
* Remove unused `nanoassert`
